minos/api_gateway/rest/handler.py

```diff
@@ -28,10 +38,29 @@ async def orchestrate(request: web.Request) -> web.Response:
 
     discovery_data = await discover(discovery_host, int(discovery_port), "/microservices", verb, url)
 
-    microservice_response = await call(**discovery_data, original_req=request)
+    user = await get_user(request)
+
+    microservice_response = await call(**discovery_data, original_req=request, user=user)
     return microservice_response
 
 
+async def get_user(request) -> str:
+    try:
+        await get_token(request)
+    except NoTokenException:
+        return ANONYMOUS
+    else:
+        try:
+            jwt_payload = await authenticate("localhost", "8082", "POST", "token", request.headers.copy())
```

Comment:
If the `authentication` microservice directly return the `user: UUID` value we'll fully isolate the authentication token protocol (`JWT` in this case) from the api gateway.

Reply:
Done.